dev-lisp/clisp/clisp-2.49.93~git.recipe

```diff
@@ -0,0 +1,90 @@
+SUMMARY="ANSI Common Lisp Implementation"
+DESCRIPTION="GNU CLISP is an ANSI Common Lisp Implementation. \
+It is known for high portability, small image size and efficient bignums."
+HOMEPAGE="https://clisp.sourceforge.io/"
+COPYRIGHT="1992-1993 Bruno Haible, Michael Stoll
+	1994-1997 Bruno Haible, Marcus Daniels
+	1998 Bruno Haible, Pierpaolo Bernardi, Sam Steingold
+	1999-2000 Bruno Haible, Sam Steingold
+	2001-2018 Sam Steingold, Bruno Haible
+	"
+LICENSE="GNU GPL v2"
+REVISION="1"
+srcGitRev="c735dd548ba2365804bcee9c2bdd3c291657e8ae"
+SOURCE_URI="https://gitlab.com/gnu-clisp/clisp/-/archive/$srcGitRev.tar.gz"
+CHECKSUM_SHA256="7c29554297e5d27abd7e50fff7691b4dc2ee26bd694369b238fb368b38278bf7"
+SOURCE_DIR="clisp-$srcGitRev"
+
+ARCHITECTURES="all !x86_gcc2"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	clisp$secondaryArchSuffix = $portVersion
+	cmd:clisp$secondaryArchSuffix = $portVersion
+	cmd:clisp_link$secondaryArchSuffix = $portVersion
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libsigsegv$secondaryArchSuffix
+	devel:libffcall$secondaryArchSuffix
+	devel:libiconv$secondaryArchSuffix
+	devel:libreadline$secondaryArchSuffix >= 8
+	devel:libintl$secondaryArchSuffix
+	devel:libunistring$secondaryArchSuffix
+	devel:libncurses$secondaryArchSuffix
+	"
+
+BUILD_PREREQUIRES="
+	cmd:awk
+	cmd:cmp
+	cmd:gcc$secondaryArchSuffix
+	cmd:make
+	cmd:msgfmt
+	"
+
+BUILD()
+{
+	./configure \
+		FORCE_UNSAFE_CONFIGURE=1 \
+		LDFLAGS="-lbsd -lnetwork" \
+		--with-included-regex \
+		--prefix=$prefix \
+		--datarootdir=$dataDir \
+		--bindir=$binDir \
+		--libdir=$libDir \
+		--docdir=$docDir \
+		--mandir=$manDir \
+		build-dir
+	# TODO Waiting for Haiku specific src/config.lisp
+	#sed -i 's|(defparameter *editor* "vi" "The name of the editor.")|(defparameter *editor* "/boot/system/apps/Pe/Pe" "The name of the editor.")|g' src/config.lisp
+	#sed -i 's|;; (setq *browser* :firefox)|;; (setq *browser* :firefox)\n(setq *browser* "/boot/system/apps/WebPositive")|g' src/config.lisp
```

Comment:
This could be done with a patch or upfront in `src/cfgunix.lisp`

Reply:
Didn''t notice these lines were commented out :) Anyway, comment stands for this (as long as there is no upstream change), you could patch the named file for now.

Reply:
Don't know how upstream is going to handle this, looking at Fedora they do it the manual way also :)

https://src.fedoraproject.org/rpms/clisp/blob/rawhide/f/clisp.spec#_156

Reply:
Unfortunately, the patch needs to be applied between configure and make: before configure the file is not existing yet, after make the change is useless, because its contents already is embedded in core image. So, `sed` until the change is available upstream (if I succeed in invoking it).

Reply:
If the file is generated by configure, is there a template file that could be patched instead, or the configure script itself?

Reply:
This will be handled upstream anyway (ongoing). Until then I managed to `sed` it up to put Haiku specific applications. For now, `cfgunix.lisp` is used as a template, I submitted `cfghaiku.lisp` upstream, but this also need some patching of `makefile.in`, which anyway needs to be handled upstream.

Just prepared and locally tested the fixed recipe.

Reply:
> Unfortunately, the patch needs to be applied between configure and make: before configure the file is not existing yet, after make the change is useless, because its contents already is embedded in core image. So, `sed` until the change is available upstream (if I succeed in invoking it).

As mentioned earlier: 
````
This could be done with a patch or upfront in src/cfgunix.lisp
````
My understanding is that `**config.lisp is generated from cfgunix.lisp**` which "is" in the source after unpacking, patching this prior to running configure as is done on the Fedora link fixes this?

Reply:
Yes, `cfgunix.lisp` is the source for `config.lisp` and it can be patched before configure. I already submitted patching through `sed`. If you consider this is quite fragile patching, I prepare and submit patched for `cfgunix.lisp` instead.

Reply:
With the latest changes to the recipe it boils down to the same. +1 :)